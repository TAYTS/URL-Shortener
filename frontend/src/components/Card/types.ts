/////////////////////////////////////////
/*           typing definition         */
/////////////////////////////////////////
import { ReactElement } from 'react';

// Component props
export interface CardProps {
  className?: string;
  children: string | ReactElement | (string | ReactElement)[];
  type?: 'default' | 'danger';
}
