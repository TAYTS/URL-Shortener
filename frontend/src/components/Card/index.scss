@import 'themes/themeVariables.scss';

.card__container {
  border-radius: 12px;
  background-color: $disabledColor;
  color: black;
  font-size: $font-size-sm;
  min-height: 50px;
  min-width: 50px;
  padding: 12px 24px;
}

.card__container--danger {
  background-color: $dangerColor;
  color: white;
}
