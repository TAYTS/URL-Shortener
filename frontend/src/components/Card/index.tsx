/////////////////////////////////////////
/*            import library           */
/////////////////////////////////////////
import React, { FunctionComponent } from 'react';
import classnames from 'classnames';

/////////////////////////////////////////
/*            import utils             */
/////////////////////////////////////////
import './index.scss';
import { CardProps } from './types';

/////////////////////////////////////////
/*            main component           */
/////////////////////////////////////////
const Card: FunctionComponent<CardProps> = ({
  className,
  children,
  type = 'default',
  dataTestId = 'card',
}: CardProps) => {
  /////////////////////////////////////////
  /*              lifecycle              */
  /////////////////////////////////////////
  return (
    <div
      className={classnames('card__container', className, {
        'card__container--danger': type === 'danger',
      })}
      data-testid={dataTestId}
    >
      {children}
    </div>
  );
};

export default Card;
