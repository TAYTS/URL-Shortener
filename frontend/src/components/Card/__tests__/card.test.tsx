/////////////////////////////////////////
/*            import library           */
/////////////////////////////////////////
import React from 'react';
import { render } from '@testing-library/react';

/////////////////////////////////////////
/*            import utils             */
/////////////////////////////////////////
import Card from '../index';

/////////////////////////////////////////
/*            test definition          */
/////////////////////////////////////////
describe('Card', () => {
  test('is rendered', () => {
    const component = render(<Card>Card Content</Card>);

    expect(component).toMatchSnapshot();
  });

  describe('props', () => {
    test('danger type', () => {
      const component = render(<Card type="danger">Card Content</Card>);

      expect(component).toMatchSnapshot();
    });

    test('additional className', () => {
      const component = render(<Card className="testing">Card Content</Card>);

      expect(component).toMatchSnapshot();
    });
  });
});
