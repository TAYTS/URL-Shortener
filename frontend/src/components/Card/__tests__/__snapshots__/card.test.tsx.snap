// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card is rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card__container"
        data-testid="card"
      >
        Card Content
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card__container"
      data-testid="card"
    >
      Card Content
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card props additional className 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card__container testing"
        data-testid="card"
      >
        Card Content
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card__container testing"
      data-testid="card"
    >
      Card Content
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card props danger type 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card__container card__container--danger"
        data-testid="card"
      >
        Card Content
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card__container card__container--danger"
      data-testid="card"
    >
      Card Content
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
